### PR TITLE
Multi-labels functionality

### PR DESCRIPTION
**Pull request by NiranjanChaudhari0929**
Creating this pull request for adding multi-label functionality to the existing code.

### CODE REVIEW

**Comment by NiranjanChaudhari0929** [comment]
can you please look into the changes ? they are final from my side.

File: src/lmcontrol/nn/clf.py [hunk 1]
```diff
@@ -36,106 +36,155 @@
...
+        return tuple(ret)
+
+    def _score(self, step_type, outputs, loss_components, true_labels):
+        ret = torch.tensor(0, device=outputs[0].device)
+        for _label_type, _output, _loss, _label in zip(self.label_classes, outputs, loss_components, true_labels):
+            _classes = self.label_classes[_label_type]
+            if _classes is None:
+                self.log(f"{step_type}_{_label_type}_loss", _loss, on_step=False, on_epoch=True)
+                r2 = r2_score(_label.cpu().detach().numpy(), _output.cpu().detach().numpy())
+                self.log(f"{step_type}_{_label_type}_r2_score", r2, on_step=False, on_epoch=True)
```
Comment (ajtritt) [inline src/lmcontrol/nn/clf.py:129]:
```suggestion
                self.log(f"{step_type}_{_label_type}_r2", r2, on_step=False, on_epoch=True)
```

**Review by ajtritt — APPROVED** [review]
